### DISCUSSION TITLE
use external resources, or allow full javascript in form.io custom validation

### DISCUSSION THREAD

**Discussion by xharkie** [formio/formio.js#4739] [Q&A]
A.

I have a field need to do validation based on external resources. But form.io doesn't allow me to use javascript fetch method and custom a variable in the custom validation.

There seems to be a data source function in premium plan, but I will host all the javascript locally, so I don't plan to upgrade the plan.

B.
There are two select fields, I want field A use external content and affects the content appeared in field B, is there any practice example?

**Comment by xharkie** [comment]
I figured that out.
But another question, how to use public google sheet as data sources?

![image](https://user-images.githubusercontent.com/107445401/173471517-a16bff93-56c8-4c46-a9b7-7930dea5253c.png)

**Reply by tanvi2808** [reply]
Hi
How did you solve problem A where I also want to call external API on custom Validation but I  am not able to do so.
I dont want to use hooks as they do api calling from render side. We want some way to do so on server side. Any help would be appreciated
